pkg/tarmak/stack/rules.go

```diff
@@ -0,0 +1,362 @@
+// Copyright Jetstack Ltd. See LICENSE for details.
+package stack
+
+import (
+	"fmt"
+)
+
+type Rules struct{}
+
+type FirewallRule struct {
+	Name        string
+	Type        string
+	FromPort    int
+	ToPort      int
+	Protocol    string
+	CIDIRBlocks []string
+}
```

Comment:
I think it would be best to add these structs:

```
type Hosts {
  Name string        // to build up a rule name programmatically 
  Role string          // to reference instance pool roles
  CIDR *net.IPnet // to use CIDr
}

type Service struct {
  Name string // service name
  Protocol tcp|udp|icmp|any
  Ports []Port
}

type Port {
 RangeFrom *unit16
 RangeTo *unit16
 Single *unit16
}
```
Then Rules would look like
```
Services []Service
Sources []Hosts
Destination []Hosts
```

And you should be able to build the name dynamically in the templates by looping over the single slices